lib/if.h

```diff
@@ -12,6 +12,10 @@
 #include "qobj.h"
 #include "hook.h"
 #include "admin_group.h"
+#ifdef GNU_LINUX
```

Comment:
The whole point of the if.h file is to provide a bit of abstraction such that FRR itself does not need to know anything about the linux internals.  *WHY* are we putting this in this header then?